main.go

```diff
@@ -1,41 +1,52 @@
 package main
 
 import (
+	"context"
 	"flag"
 	"fmt"
 	"log"
 	"net/http"
 	"os"
+	"os/signal"
+	"syscall"
 	"time"
 
 	"gopkg.in/yaml.v2"
 )
 
-// Define var for config path
-var configPath string
-
 // Config struct for webapp config
 type Config struct {
 	Server struct {
-		Host    string `yaml:"host"`
+		// Host is the local machine IP Address to bind the HTTP Server to
+		Host string `yaml:"host"`
+
+		// Port is the local machine TCP Port to bind the HTTP Server to
 		Port    string `yaml:"port"`
 		Timeout struct {
-			Write int `yaml:"write"`
-			Read  int `yaml:"read"`
-			Idle  int `yaml:"idle"`
+			// Server is the general server timeout to use for graceful shutdowns
+			Server time.Duration `yaml:"server"`
```

Comment:
Rather than having to read these fields in as integers, you can just read them in as time.Duration's so you don't have to cast them elsewhere (unless you plan to use these values as anything other than time.Duration, which is doubtful in the context).